From c05b6ce21ed7a01183442056f6893d06ff0f6848 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Mon, 5 Feb 2024 11:22:55 +0000
Subject: [PATCH 714/748] staging: bcm2835-codec: Fix up for 6.8 - use
 ignore_cap_streaming

Drops downstream patch to v4l2_mem2mem, and uses the new mainline
flag to achieve the same functionality

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 .../staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c    | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c b/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
index cafde80d094b..c6803093c985 100644
--- a/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
+++ b/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
@@ -3522,6 +3522,8 @@ static int bcm2835_codec_open(struct file *file)
 	v4l2_m2m_set_src_buffered(ctx->fh.m2m_ctx, true);
 	v4l2_m2m_set_dst_buffered(ctx->fh.m2m_ctx, true);
 
+	ctx->fh.m2m_ctx->ignore_cap_streaming = true;
+
 	v4l2_fh_add(&ctx->fh);
 	atomic_inc(&dev->num_inst);
 
-- 
2.43.0

